src/core.rs

```diff
@@ -1290,6 +1304,82 @@ mod tests {
+            None,
+        );
+
+        // Generated by a yubico 5
+        // Make a "fake" challenge, where we know what the values should be ....
+
+        let zero_chal = Challenge::new(vec![
+            160, 127, 213, 174, 150, 36, 228, 190, 41, 61, 216, 14, 171, 191, 75, 203, 99, 59, 4,
+            252, 49, 90, 235, 36, 220, 165, 159, 201, 58, 225, 248, 142,
+        ]);
+
+        // Create the fake credential that we know is associated
+        let cred = Credential {
+            counter: 1,
+            cred_id: vec![
+                179, 64, 237, 0, 28, 248, 197, 30, 213, 228, 250, 139, 28, 11, 156, 130, 69, 242,
+                21, 48, 84, 77, 103, 163, 66, 204, 167, 147, 82, 214, 212,
+            ],
+            cred: COSEKey {
+                type_: COSEAlgorithm::ES256,
+                key: COSEKeyType::EC_EC2(COSEEC2Key {
+                    curve: ECDSACurve::SECP256R1,
+                    x: [
+                        187, 71, 18, 101, 166, 110, 166, 38, 116, 119, 74, 4, 183, 104, 24, 46,
+                        245, 24, 227, 143, 161, 136, 37, 186, 140, 221, 228, 115, 81, 175, 50, 51,
+                    ],
+                    y: [
+                        13, 59, 59, 158, 149, 197, 116, 228, 99, 12, 235, 185, 190, 110, 251, 154,
+                        226, 143, 75, 26, 44, 136, 244, 245, 243, 4, 40, 223, 22, 253, 224, 95,
+                    ],
+                }),
+            },
+            verified: false,
+            registration_policy: UserVerificationPolicy::Discouraged,
+        };
+
+        // Persist it to our fake db.
+        let wan = Webauthn::new(wan_c);
+
+        // Captured authentication attempt
+        let rsp = r#"
+        {
+            "id": "z077A6SzdvA3rDRG6tfnTf9TMFVtfLYe1mh27mRXgBZU6TXA_nCJAi6WnLLq1p3d0yj3n62_4yJMu80o4O8kkw",
+            "rawId": "z077A6SzdvA3rDRG6tfnTf9TMFVtfLYe1mh27mRXgBZU6TXA_nCJAi6WnLLq1p3d0yj3n62_4yJMu80o4O8kkw",
+            "type": "public-key",
+            "extensions": {
+                "appid": true
+            },
+            "response": {
+                "authenticatorData": "UN6WJxNDzSGdqrQoPbqYbsZdIxtC9vfU9iGe5i1pCTYBAAAAuQ",
+                "clientDataJSON": "eyJjaGFsbGVuZ2UiOiJvSF9WcnBZazVMNHBQZGdPcTc5THkyTTdCUHd4V3VzazNLV2Z5VHJoLUk0IiwiY2xpZW50RXh0ZW5zaW9ucyI6eyJhcHBpZCI6Imh0dHBzOi8vdGVzdGluZy5sb2NhbC9hcHAtaWQuanNvbiJ9LCJoYXNoQWxnb3JpdGhtIjoiU0hBLTI1NiIsIm9yaWdpbiI6Imh0dHBzOi8vdGVzdGluZy5sb2NhbCIsInR5cGUiOiJ3ZWJhdXRobi5nZXQifQ",
+                "signature": "MEUCIEw2O8LYZj6IKbjP6FuvdcL2MoDBY6LqJWuhteje3H7eAiEAvzRLSg70tkrGnZqpQIZyv-zaizNpCtyr4U3SZ-E2-f4"
+            }
+        }
+        "#;
+        let rsp_d: PublicKeyCredential = serde_json::from_str(rsp).unwrap();
+
+        // Now verify it!
+        let r = wan.verify_credential_internal(
+            &rsp_d,
+            UserVerificationPolicy::Discouraged,
+            &zero_chal,
+            &cred,
+            &Some(String::from("https://testing.local/app-id.json")),
```

Comment:
Probably also worth a duplication of this verify_credential_internal where Option<Appid> is None to show it fails when you don't have appid with this credential response. IE .is_err().